Tests fail due to missing PyAV dependency on dev installation

The following tests fail:
```
FAILED tests/data/test_LightlyDataset.py::TestLightlyDataset::test_video_dataset_filenames - NameError: name 'cv2' is not defined
FAILED tests/data/test_LightlySubset.py::TestLightlyDataset::test_video_dataset_filenames - NameError: name 'cv2' is not defined
FAILED tests/data/test_LightlySubset.py::TestLightlySubset::test_video_dataset_filenames - NameError: name 'cv2' is not defined
FAILED tests/data/test_VideoDataset.py::TestVideoDataset::test_video_dataset_from_folder - ImportError: PyAV is not installed, and is necessary fo...
FAILED tests/data/test_VideoDataset.py::TestVideoDataset::test_video_similar_timestamps_for_different_backends - ImportError: PyAV is not installe...
```
The first 3 tests fail because no check is performed to see if cv2 and PyAV are loaded. The 2 remaining tests fail because  torchvision is imported and a VideoDataset created without checking whether PyAV is installed.

--- a/tests/data/test_LightlyDataset.py
+++ b/tests/data/test_LightlyDataset.py
@@ -297,6 +297,7 @@ class TestLightlyDataset(unittest.TestCase):
         for filename in os.listdir(out_dir):
             self.assertIn(filename, dataset.get_filenames()[(len(dataset) // 2):])
 
+    @unittest.skipUnless(VIDEO_DATASET_AVAILABLE, "PyAV or CV2 are not available")
     def test_video_dataset_filenames(self):
         self.create_video_dataset()
         all_filenames = self.filenames

--- a/tests/data/test_LightlyDataset.py
+++ b/tests/data/test_LightlyDataset.py
@@ -268,26 +268,8 @@ class TestLightlyDataset(unittest.TestCase):
                 self.assertIsInstance(filename, str)
                 self.assertIn(filename, filenames_dataset)
 
-    def test_video_dataset(self):
-
-        if not VIDEO_DATASET_AVAILABLE:
-            tmp_dir = tempfile.mkdtemp()
-            # simulate a video
-            # the video dataset will check to see whether there exists a file
-            # with a video extension, it's enough to fake a video file here
-            path = os.path.join(tmp_dir, 'my_file.png')
-            dataset = torchvision.datasets.FakeData(size=1, image_size=(3, 32, 32))
-            image, _ = dataset[0]
-            image.save(path)
-            os.rename(path, os.path.join(tmp_dir, 'my_file.avi'))
-            with self.assertRaises(ImportError):
-                dataset = LightlyDataset(input_dir=tmp_dir)
-
-            warnings.warn(
-                'Did not test video dataset because of missing requirements')
-            shutil.rmtree(tmp_dir)
-            return
-
+    @unittest.skipUnless(VIDEO_DATASET_AVAILABLE, "PyAV and CV2 are both installed")
+    def test_video_dataset_available(self):
         self.create_video_dataset()
         dataset = LightlyDataset(input_dir=self.input_dir)
 
@@ -296,6 +278,23 @@ class TestLightlyDataset(unittest.TestCase):
         self.assertEqual(len(os.listdir(out_dir)), len(dataset) // 2)
         for filename in os.listdir(out_dir):
             self.assertIn(filename, dataset.get_filenames()[(len(dataset) // 2):])
+    
+    @unittest.skipIf(VIDEO_DATASET_AVAILABLE, "PyAV or CV2 is/are not installed")
+    def test_video_dataset_unavailable(self):
+        tmp_dir = tempfile.mkdtemp()
+        # simulate a video
+        # the video dataset will check to see whether there exists a file
+        # with a video extension, it's enough to fake a video file here
+        path = os.path.join(tmp_dir, 'my_file.png')
+        dataset = torchvision.datasets.FakeData(size=1, image_size=(3, 32, 32))
+        image, _ = dataset[0]
+        image.save(path)
+        os.rename(path, os.path.join(tmp_dir, 'my_file.avi'))
+        with self.assertRaises(ImportError):
+            dataset = LightlyDataset(input_dir=tmp_dir)
+
+        shutil.rmtree(tmp_dir)
+        return
 
     @unittest.skipUnless(VIDEO_DATASET_AVAILABLE, "PyAV or CV2 are not available")
     def test_video_dataset_filenames(self):

--- a/tests/data/test_LightlyDataset.py
+++ b/tests/data/test_LightlyDataset.py
@@ -22,7 +22,7 @@ try:
     import cv2
 
     VIDEO_DATASET_AVAILABLE = True
-except Exception:
+except ModuleNotFoundError:
     VIDEO_DATASET_AVAILABLE = False
 
 

--- a/tests/data/test_VideoDataset.py
+++ b/tests/data/test_VideoDataset.py
@@ -13,7 +13,7 @@ import cv2
 try:
     import av
     PYAV_AVAILABLE = True
-except:
+except ModuleNotFoundError:
     PYAV_AVAILABLE = False
 
 class TestVideoDataset(unittest.TestCase):

--- a/tests/data/test_VideoDataset.py
+++ b/tests/data/test_VideoDataset.py
@@ -10,7 +10,17 @@ import torchvision
 from lightly.data._video import VideoDataset, _make_dataset
 import cv2
 
+try:
+    import av
+    PYAV_AVAILABLE = True
+except:
+    PYAV_AVAILABLE = False
+
 class TestVideoDataset(unittest.TestCase):
+
+    def setUp(self):
+        if not PYAV_AVAILABLE:
+            self.skipTest('PyAV not available')
 
     def ensure_dir(self, path_to_folder: str):
         if not os.path.exists(path_to_folder):

